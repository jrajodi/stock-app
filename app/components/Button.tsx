import * as React from "react";
import { View, StyleSheet, Text } from "react-native";

interface Props {
  status: number;
}

interface State {}

export default class Button extends React.PureComponent<Props, State> {
  constructor(props: any) {
    super(props);
  }

  render() {
    const { status = 0 } = this.props;

    let text = "BUY",
      style = styles.buy;
    if (status == 1) {
      text = "SELL";
      style = styles.sell;
    }
    if (status == 2) {
      text = "HOLD";
      style = styles.hold;
    }

    return (
      <View style={[styles.bshButton, style]}>
        <Text style={[styles.text, styles.white]}>{text}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  bshButton: {
    padding: 2,
    paddingLeft: 7,
    paddingRight: 7,
    borderRadius: 10
  },
  text: {
    fontFamily: "BrandonText-Medium",
    fontSize: 14,
    color: "#3d3d3d"
  },
  buy: {
    backgroundColor: "#00AF64"
  },
  sell: {
    backgroundColor: "#F56A42"
  },
  hold: {
    backgroundColor: "#FECD64"
  },
  white: {
    color: "#FFF"
  }
});
