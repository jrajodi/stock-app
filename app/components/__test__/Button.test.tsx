// const React = require("react");
// const Button = require("../Button").default;
// const renderer = require("react-test-renderer");

import Button from "../Button";
import renderer from "react-test-renderer";
import React from "react";


test("renders correctly", () => {
  const tree = renderer.create(<Button status={0} />).toJSON();
  expect(tree).toMatchSnapshot();
});
