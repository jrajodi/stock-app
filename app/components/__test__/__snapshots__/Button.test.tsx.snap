// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 10,
        "padding": 2,
        "paddingLeft": 7,
        "paddingRight": 7,
      },
      Object {
        "backgroundColor": "#00AF64",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#3d3d3d",
          "fontFamily": "BrandonText-Medium",
          "fontSize": 14,
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    BUY
  </Text>
</View>
`;
